### ISSUE TITLE
Harmonize unique and primary_key?

### ISSUE THREAD

**Issue by sveinugu** [fairtracks/fairtracks_validator#10]
I cannot see any difference in the README between unique and primary_key keywords. However, the implementation code is different? Perhaps this is due to one of them being more updated than the other (I assume "unique")? Is there any difference?

Should one of these be deprecated, and just remain as an alias to the other for backwards compatibility? As "primary_key" might have more technical debt connected to it, I have no problems with deprecating "unique" and update the FAIRtracks schema accordingly.

**Comment by sveinugu** [comment]
Of course, one should check the two implementations and make sure the most up-to-date/correct code is used further on.

**Comment by jmfernandez** [comment]
(Primary key)[https://en.wikipedia.org/wiki/Primary_key] and (unique keys)[https://en.wikipedia.org/wiki/Unique_key] concepts come from relational models . All the unique keys are candidates to primary key, but there can be at most a primary key. In relational models, the value of the primary key from a tuple (in this case, a document) is used to keep a reference from other tuples / documents where foreign key definitions are found.

Therefore, the constraints primary key and unique keys define are different, so they are not redundant. And the validator itself has been designed from the very beginning to be reused in very different scenarios.

**Comment by sveinugu** [comment]
Right, I see.

I don't think I had the difference in mind when suggesting the `unique` keyword, and as far as I can see, all the `unique` keys in FAIRtracks are conceptually primary keys (`local_id`), and could be denoted as such. As far as I can see, the only difference from the `primary_key` implementation is that there can be more than one `unique` property in the same object, but this is not needed for FAIRtracks, I think. There is also the difference in the implementation of `foreignProperty` as compared to `foreign_key`, in that the former can refer to JSON parts within a schema, while the latter can only refer to a top-level property in a (sub)schema. I don't know what possibilities/complications arise from this, so I don't know which concept is the most fruitful to follow, if one were to deprecate the other.  But I think FAIRtracks can be easily rearranged to make use of the primary_key/foreign_key extension. I think this whole split really only happened because I did not have the full overview of the existing primary_key/foreign_key logic in the validator when I suggested the unique/foreignProperty solution. I am happy to continue with it, as it allows for the `doc_info` object (which again contains a `unique` `local_id` property) to be included in the top-level FAIRtracks JSON (where I feel it belongs) and not in a separate `doc_info` subschema, but having to move it would not mean the end of the world.

Or am I missing something?

In any case, I think it would be advantageous to improve the README somewhat to more clearly explain the differences in logic between the two mechanisms.
